lstchain/irf/hdu_table.py

```diff
@@ -0,0 +1,374 @@
+            "OBJECT": event_table.meta["OBJECT"],
+            "OBS_MODE": event_table.meta["OBS_MODE"],
+            "N_TELS": event_table.meta["N_TELS"],
+            "TELLIST": event_table.meta["TELLIST"],
+        }
+        obs_index_tables.append(t_obs)
+
+    hdu_index_table = Table(hdu_index_tables)
+
+    hdu_index_header = DEFAULT_HEADER.copy()
+    hdu_index_header["HDUCLAS1"] = "INDEX"
+    hdu_index_header["HDUCLAS2"] = "HDU"
+    hdu_index_header["TELESCOP"] = "CTA"
+    hdu_index_header["INSTRUME"] = f"{t_obs['TELLIST']}"
+
+    hdu_index = fits.BinTableHDU(
+        hdu_index_table, header=hdu_index_header, name="HDU INDEX"
+    )
+    hdu_index_list = fits.HDUList([fits.PrimaryHDU(), hdu_index])
+
+    obs_index_table = QTable(obs_index_tables)
+
+    obs_index_header = hdu_index_header.copy()
+    obs_index_header["HDUCLAS2"] = "OBS"
+    obs_index_header["MJDREFI"] = event_table.meta["MJDREFI"]
+    obs_index_header["MJDREFF"] = event_table.meta["MJDREFF"]
+
+    obs_index = fits.BinTableHDU(
+        obs_index_table, header=obs_index_header, name="OBS INDEX"
+    )
+    obs_index_list = fits.HDUList([fits.PrimaryHDU(), obs_index])
+
+    return hdu_index_list, obs_index_list
+
+
+def create_event_list(data, run_number, source_name):
+    """
+    Create the event_list BinTableHDUs from the given data
+
+    Parameters
+    ----------
+        Data: DL2 data file
+                'astropy.table.QTable'
+        Run: Run number
+                Int
+        Source_name: Name of the source
+                Str
+    Returns
+    -------
+        Events HDU:  `astropy.io.fits.BinTableHDU`
+        GTI HDU:  `astropy.io.fits.BinTableHDU`
+        Pointing HDU:  `astropy.io.fits.BinTableHDU`
+    """
+
+    tel_list = np.unique(data["tel_id"])
+
+    # Timing parameters
+    t_start = data["dragon_time"].value[0]
+    t_stop = data["dragon_time"].value[-1]
+    time = Time(data["dragon_time"], format="unix", scale="utc")
+    date_obs = time[0].to_value("iso", "date")
+    obs_time = t_stop - t_start
+    # using method from PR#566
+    deltaT = np.diff(data["dragon_time"].value)
+    deltaT = deltaT[(deltaT > 0) & (deltaT < 0.002)]
+    rate = 1 / np.mean(deltaT)
+    dead_time = np.amin(deltaT)
+
+    # Position parameters
+    location = EarthLocation.from_geodetic(
+        -17.89139 * u.deg, 28.76139 * u.deg, 2184 * u.m
+    )
+    reco_alt = data["reco_alt"]
+    reco_az = data["reco_az"]
+    pointing_alt = data["pointing_alt"]
+    pointing_az = data["pointing_az"]
+
+    src_sky_pos = SkyCoord(
+        alt=reco_alt, az=reco_az, frame=AltAz(obstime=time, location=location)
+    ).transform_to(frame="icrs")
+    tel_pnt_sky_pos = SkyCoord(
+        alt=pointing_alt[0],
+        az=pointing_az[0],
+        frame=AltAz(obstime=time[0], location=location),
+    ).transform_to(frame="icrs")
+
+    try:
+        object_radec = SkyCoord.from_name(source_name)
+    except Exception:
```

Comment:
`TimeoutException` instead?

Reply:
Here there is an `astropy.coordinates.name_resolve.NameResolveError` that occurs when Sesame cannot resolve the name provided, So I cannot use a Timeout error/exception.

Reply:
I can change the Error message to 'Name resolve error...', so that it is not confusing.

Reply:
We should not rely on this naming resolution. Do we not have the ra/dec coordinates of the source?

Reply:
Here I use the standard ra/dec information to check if there is any mispointing with respect to the 3 observation modes.
For that reason, I think it is better to compare the ra/dec values in the data file, with a standard source of information.

Reply:
This does not change any major value in the DL3, just helps in reading the observation mode and could help in checking for the 'quality' of the data, as it might be required for some high-level analyses.